apps/web/components/Organization/Wrapper/OrganizationWrapper.tsx

```diff
@@ -285,15 +284,27 @@ export const OrganizationWrapper: React.FC<WrapperProps> = ({
   showSecondaryMenu = true,
   namespace,
 }) => {
-  const Router = useRouter()
+  const router = useRouter()
 
   const secondaryNavList: NavigationItem[] =
     organizationPage.secondaryMenu?.childrenLinks.map(({ text, url }) => ({
       title: text,
       href: url,
-      active: Router.asPath === url,
+      active: router.asPath === url,
     })) ?? []
 
+  // Keep track of what navigation link is active and update it when the browser url changes
+  const activeNavigationItem = useMemo(() => {
```

Comment:
seems like with this approach you keep iterating over the whole list and children even if you've found the corresponding item.

maybe a normal `for (const x of z) {` loop would be better suited for the early exit?

also, should `activeNavigationItem` be named `activeNavigationItemTitle`

Reply:
Good point. This has been addressed accordingly.